sdks/java/container/build.gradle

```diff
@@ -101,16 +84,44 @@ docker {
                   project.rootProject["docker-tag"] : project.sdk_version)
   dockerfile project.file("./${dockerfileName}")
   files "./build/"
+  buildArgs(['pull_licenses': !project.rootProject.hasProperty(["no-licenses"])])
```

Comment:
How about we call both flags 'pull_licenses' and add "pull_licenses=true" to project root, with a comment what this flag does, and how to disable it?

Reply:
Currently, there are three pull licenses options for Java.
when docker-pull-licenses is set, licenses are pulled.
when no-licenses is set, all license pulling related tasks are skipped.
when no tag is passed, which is default mode, license urls are checked.

With Python, there are two options available.
with no-licenses tag, license pulling is skipped.
when no tag is passed, which is default, licenses are pulled.

I agree above settings are confuse, and would like to simplify it and make it consistent between docker images.

So, we will have only one tag, which is docker-pull-licenses.
when docker-pull-licenses = 1 or true, licenses are pulled.
when docker-pull-licenses = 0 or false, license related tasks are skipped.
when docker-pull-licenses = check, license urls are checked.
Default mode is docker-pull-licenses = check both for Java and Python.

We need to update the script that pull licenses for Python to support checking urls.

Does this sound good?

Reply:
Sounds good. Sorry if this was covered in the discussion that I didn't read, but why would we want to check urls but not pull? Feel free to point me to the discussion.

Naming suggestion:
licenses-in-docker-images=add
licenses-in-docker-images=skip
licenses-in-docker-images=check_urls

Reply:
In short, users want to create a lightweight images, without adding licenses.
Lightweight images are welcomed for Jenkins test as well, it reduces image size by 85MB for Java image. More discussion can be found at [here](https://lists.apache.org/thread.html/rff9f05e08de6adf7c39c0f4c59f97ae1a2f3602768480fe9e31e0428%40%3Cdev.beam.apache.org%3E)

Naming suggestion sounds good to me, thank you.

Reply:
Right, so to produce lightweight images we can skip license download. Why would we ever want to 'check' urls but not downloading the licenses? From your comment:
```
I tried to check if urls are valid instead of pulling the files and it
reduced only 1 min of running time. So, it's not an option here.
```

Reply:
The purpose of checking urls was to make sure files can be pulled from the urls when create images. If we set it to skip, no url checking process happened for the PRs and we may see many license issues when we create release images. Then release managers need to fix the issues. If we add the checking urls for each PR, contributors should fix the issues if any and will not have to fix them during release.

Reply:
There are some more discussion at the thread today, in case you haven't seen it yet.